Sources/Classes/Views/BaseCell/BaseCell.swift

```diff
@@ -0,0 +1,170 @@
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit
+import TableKit
+
+/// Base cell that provides separator support
+open class BaseCell: UITableViewCell {
+
+    // MARK: - Public
+
+    /// Configure separator with viewModel
+    /// - parameter viewModel: ViewModel of cell, that inherits from BaseCellViewModel
+    public func configureSeparator(with viewModel: BaseCellViewModel) {
+        configureInterface(with: viewModel)
+    }
+
+    // MARK: - Private
+
+    private var topView: UIView!
+    private var bottomView: UIView!
+
+    // top separator constraints
+    private var topViewLeftConstraint: NSLayoutConstraint!
+    private var topViewRightConstraint: NSLayoutConstraint!
+    private var topViewTopConstraint: NSLayoutConstraint!
+    private var topViewHeightConstraint: NSLayoutConstraint!
+
+    // bottom separator constraints
+    private var bottomViewLeftConstraint: NSLayoutConstraint!
+    private var bottomViewRightConstraint: NSLayoutConstraint!
+    private var bottomViewBottomConstraint: NSLayoutConstraint!
+    private var bottomViewHeightConstraint: NSLayoutConstraint!
+
+    private var topSeparatorInsets    = UIEdgeInsets.zero
+    private var bottomSeparatorInsets = UIEdgeInsets.zero
+
+    private var topSeparatorHeight    = CGFloat(pixels: 1)
+    private var bottomSeparatorHeight = CGFloat(pixels: 1)
+
+    // MARK: - Initialization
+
+    public required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+
+        configureLineViews()
+    }
+
+    public override init(style: UITableViewCellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+        configureLineViews()
+    }
+
+    override open func updateConstraints() {
+        topViewTopConstraint.constant       = topSeparatorInsets.top
+        topViewLeftConstraint.constant      = topSeparatorInsets.left
+        topViewRightConstraint.constant     = topSeparatorInsets.right
+        topViewHeightConstraint.constant    = topSeparatorHeight
+
+        bottomViewLeftConstraint.constant   = bottomSeparatorInsets.left
+        bottomViewRightConstraint.constant  = bottomSeparatorInsets.right
+        bottomViewBottomConstraint.constant = bottomSeparatorInsets.bottom
+        bottomViewHeightConstraint.constant = bottomSeparatorHeight
+
+        super.updateConstraints()
+    }
+
+    private func configureInterface(with viewModel: BaseCellViewModel?) {
```

Comment:
I see only one call of configureInterface (https://github.com/TouchInstinct/LeadKit/pull/91/files#diff-efde530aacfa25f2d97941c1931256b0R34) with non-optional view model